src/transformation/visitors/variable-declaration.ts

```diff
@@ -29,7 +29,7 @@ export function transformArrayBindingElement(
 export function transformBindingPattern(
     context: TransformationContext,
     pattern: ts.BindingPattern,
-    table: lua.Identifier,
+    table: lua.Expression,
```

Comment:
Seemed like no reason it was an identifier, also it makes sense this is an expression because you can also more complex expressions representing some destructible object.